### PR TITLE
Test on Python 3.11

### PR DESCRIPTION
**Pull request by mattwthompson**
- [x] Add NAGL back if its recipe is split (somehow this works now?)
- [x] Think about if there's an easier way to workaround Interchange adhering to the toolkit's packaged Python pin
- [ ] ~~Update examples environments/workflow~~ already only running on only one Python version, can do this if desired

### CODE REVIEW

File: devtools/conda-envs/test_env.yaml
Comment (mattwthompson) [inline devtools/conda-envs/test_env.yaml:23]: Ref https://github.com/conda-forge/openff-nagl-feedstock/issues/6

File: devtools/conda-envs/openeye.yaml
Comment (j-wags) [inline devtools/conda-envs/openeye.yaml:25]: Am I correct in thinking that all the changes to this file would be reverted before merging?

**Comment by j-wags** [comment]
The cost/benefit of complicating  the testing infrastructure to sidestep the unavailability of AT on python 3.11 here doesn't seem worth it. Once AT is available for py3.11 I'll be happy to enable this testing and update the package recipe.

**Comment by mattwthompson** [comment]
The complexities here are because of how the toolkit is packaged, not because of AmberTools.

This will at least assess whether or not the test suite passes with 3.11

**Comment by mattwthompson** [comment]
Thanks to unspecified magic happening upstream, this is actually working without much complexity!

**Review by j-wags — APPROVED** [review]
Approved pending tests passing! (Excluding RTD)

**Pull request merged**
